### DISCUSSION TITLE
[dove] Additional properties in user?

### DISCUSSION THREAD

**Discussion by jondmoon** [feathersjs/feathers#2999] [General]
I am trying to set up a dove application, and trying to set up my authentication through LDAP like I had set with v4. The main issue I have now is that the userSchema which is what is used in all of the service calls just has _id, email, and an optional password. In v4 I was able to store additional fields in the user object (like username, roles, etc) that was all pulled through the LDAP connection, but I cannot figure out how to get that to work in v5.

Is there a reason why the userSchema is set to not allow additional fields?

Thanks.

Jon

**Comment by daffl** [comment]
You can add any fields you need in the [`user.schema` file](https://feathersjs.com/guides/cli/service.schemas.html). Additional fields are not allowed by default for type safety and security reasons.

**Reply by jondmoon** [reply]
Okay, I feel dumb. Thanks!

Jon

**Reply by daffl** [reply]
It was a valid question. One other thing to note for SQL databases is that each schema change needs a [corresponding migration](https://feathersjs.com/guides/cli/knexfile.html)
